libs/utils/trace_analysis.py

```diff
@@ -966,7 +966,35 @@ def _integrate_square_wave(self, sq_wave):
         else:
             return sum(comp_sig.iloc[1::2].index - comp_sig.iloc[:-1:2].index)
 
-    @memoized
+    def _cropToXTimeRange(self, data):
+        """
+        Crop a dataframe containing FTrace events to the X time range specified
+        by the user through setXTimeRange().
+
+        When cropping a series it might be necessary to include a first and
+        last element with timestamps equal to the boundaries of the X time
+        range. Because of this, it is necessary to check whether timestamps
+        with those particular values already exist in the series, in order to
+        avoid duplicates in the resulting cropped data.
+
+        :param data: dataframe to be cropped
+        :type data: :mod:`pandas.Series`
+        """
+        if not isinstance(data, pd.Series):
+            raise ValueError('cropping supported only for pandas.Series '
+                             'objects!')
+
+        first = None
+        if self.x_min > 0.0 and self.x_min not in data.index:
+            first = pd.Series([data[data.index < self.x_min].iloc[-1]],
+                              index=[self.x_min])
+        last = None
+        if self.x_max < self.trace.time_range and self.x_max not in data.index:
```

Comment:
Mmm.. wondering if this check on "self.x_max < self.trace.time_range" is valid also for "non normalized" traces.
Did you tried to set a range for a non normalized trace?